tt_metal/impl/dispatch/kernels/command_queue_producer.cpp

```diff
@@ -57,12 +58,16 @@ void setup_issue_queue_read_interface(const uint32_t issue_region_rd_ptr, const
     cq_read_interface.issue_fifo_size = issue_region_size >> 4;
     cq_read_interface.issue_fifo_limit = (issue_region_rd_ptr + issue_region_size) >> 4;
     cq_read_interface.issue_fifo_rd_toggle = 0;
+    DPRINT << "RD PTR: " << issue_region_rd_ptr << ENDL();
 }
 
 void kernel_main() {
-    constexpr uint32_t host_issue_queue_read_ptr_addr = get_compile_time_arg_val(0);
-    constexpr uint32_t issue_queue_start_addr = get_compile_time_arg_val(1);
-    constexpr uint32_t issue_queue_size = get_compile_time_arg_val(2);
+    DPRINT << "STARTING PRODUCER CODE" << ENDL();
+    uint32_t host_issue_queue_read_ptr_addr = get_compile_time_arg_val(0);
+    uint32_t issue_queue_start_addr = get_compile_time_arg_val(1);
+
+    // Only the issue queue size is a runtime argument
+    uint32_t issue_queue_size = get_arg_val<uint32_t>(0);
```

Comment:
why isn't the completion queue size a rt arg in the command_queue_consumer?

Reply:
Because it never changes.

Reply:
if the issue queue size increases wouldn't completion queue size reduce

Reply:
The issue queue size is not increasing. The new size is <= issue queue size.

Reply:
ah okay, maybe I missed that